### DISCUSSION TITLE
How to send a reply with a RichtextFacet.Link

### DISCUSSION THREAD

**Discussion by basinbasin** [MarshalX/atproto#175] [Questions] [answered]
I can see in the examples how to send either one of these things, but I can't work out how to send a reply that also includes a clickable url in the reply text. Thank you.

**Comment by Vadimkin** [comment]
More likely `client.send_post()` can be extended with `facets` and `reply_ro` arguments...

As some solution, I copy-pasted everything from `send_post()` into my own method and added `facets` + `reply_to` arguments method to the `models.AppBskyFeedPost.Main` instance.

Pseudocode:
```
reply_to = models.AppBskyFeedPost.ReplyRef(parent=..., root=...)
facets = [...]

record_data = models.ComAtprotoRepoCreateRecord.Data(
    repo=repo,
    collection=models.ids.AppBskyFeedPost,
    record=models.AppBskyFeedPost.Main(
        created_at=datetime.now().isoformat(), text=message, reply=reply_to, facets=facets, embed=embed, langs=["en"]
    ),
)
```

**Comment by MarshalX** [comment, marked as answer]
Now you can use https://atproto.blue/en/latest/text_builder.html for this. Also, the facets parameter has been added to the high-level API (send_post method)
